### PR TITLE
Rework and expand QueryPhaseTests

### PR DESCRIPTION
**Pull request by javanna**
There are a number of issues with the current tests included in QueryPhaseTests that this commit addresses:
- many scenarios tested in the same single test method that can be split up into multiple test methods
- the search context is shared across subsequent test sections, and modified for each scenarios, which makes it hard to follow exactly what scenario each section tests
- some scenarios are tested multiple times
- code duplication around indexing docs

The added code coverage revolves around support for terminate_after, which was only tested in combination with the scenario where total hit count can be shortcut. Added scenario where it cannot be shortcut.

### CODE REVIEW

**Review by mayya-sharipova — APPROVED** [review]
@javanna Thanks Luca, the test suite indeed looks much organized with your changes.

**Comment by javanna** [comment]
run elasticsearch-ci/bwc
